File: E3/E03-01/app/src/main/java/com/example/e03_01/MainActivity.java
package com.example.e03_01;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.example.e03_01.adapters.EmployeesAdapter;
import com.example.e03_01.enums.Gender;
import com.example.e03_01.enums.Role;
import com.example.e03_01.viewmodels.Employee;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EmployeesAdapter employeesAdapter = new EmployeesAdapter();
        RecyclerView recyclerView = findViewById(R.id.recycler_view);

        recyclerView.setAdapter(employeesAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this, RecyclerView.VERTICAL, false));

        List<Employee> employeeList = new ArrayList<Employee>(Arrays.asList());
        Employee newEmployee = new Employee();
        newEmployee.setName("Armend");
        newEmployee.setGender(Gender.Male);
        newEmployee.setRole(Role.Technology);
        employeeList.add(newEmployee);
        employeesAdapter.setData(employeeList);
    }
}

File: 2023-SOPM/ClassStudentManagement/app/src/main/java/com/example/classstudentmanagement/MainActivity.kt
package com.example.classstudentmanagement

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    val TAG = "MainActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d(TAG, "Called onCreate")

        val button = findViewById<Button>(R.id.login_button)
        val username = findViewById<EditText>(R.id.username)
        val password = findViewById<EditText>(R.id.password)
        button.setOnClickListener {
            if (username.text.isNullOrEmpty() || password.text.isNullOrEmpty()) {
                Toast
                    .makeText(this, "Username or Password is empty", Toast.LENGTH_LONG)
                    .show()
                return@setOnClickListener
            }

            val intent = Intent(this, ContentActivity::class.java)
            intent.putExtra("USERNAME", username.text.toString())
            startActivity(intent)
            finish()
        }
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "Called onStart")
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "Called onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "Called onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "Called onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "Called onDestroy")
    }
}